src/Middleware/HeaderPropagation/samples/HeaderPropagationSample/HeaderPropagationSample/Program.cs

```diff
@@ -0,0 +1,21 @@
+using System.IO;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.Hosting;
+
+namespace HeaderPropagationSample
+{
+    public class Program
+    {
+        public static void Main(string[] args)
+        {
+            var host = new WebHostBuilder()
```

Comment:
We need to get into the habit of having samples use generic host, CreateDefaultBuilder, etc.. We no longer have cross-repo layering restrictions writing them, so they should all look like the templates.

Reply:
Cool beans.

Reply:
So I'm trying to follow your advice on this. I have currently ended up in a pit of failure that I am already familiar with

1. Adding `ConfigureWebHostDefaults` requires `Microsoft.AspNetCore`
2. Adding `Microsoft.AspNetCore` requires `Microsoft.AspNetCore.Server.IIS`
3. Adding `Microsoft.AspNetCore.Server.IIS` results in:

```
2>C:\Program Files (x86)\Microsoft Visual Studio\d16.1stg\MSBuild\Current\Bin\Microsoft.Common.CurrentVersion.targets(4552,5): error MSB3030: Could not copy the file "C:\git\aspnet\AspNetCore\src\Servers\IIS\AspNetCoreModuleV2\InProcessRequestHandler\bin\Debug\Win32\aspnetcorev2_inprocess.dll" because it was not found.
2>C:\Program Files (x86)\Microsoft Visual Studio\d16.1stg\MSBuild\Current\Bin\Microsoft.Common.CurrentVersion.targets(4552,5): error MSB3030: Could not copy the file "C:\git\aspnet\AspNetCore\src\Servers\IIS\AspNetCoreModuleV2\InProcessRequestHandler\bin\Debug\x64\aspnetcorev2_inprocess.dll" because it was not found.
2>Done building project "Microsoft.AspNetCore.Server.IIS.csproj" -- FAILED.
```

I thought we fixed this?

Reply:
That's odd, it's working in my sample
https://github.com/aspnet/AspNetCore/pull/9831/files#diff-55f73ec5609ad0e27d098fa76a7e32a4R17

Have you run build on the repo recently?
@jkotalik

Reply:
I have other solutions that use `Microsoft.AspNetCore.Servers.IIS` that are working fine. I'll try building and see if that fixes it.

Reply:
So I did a build, and did a bunch of things and I can't get this to work. The `AuthSamples.sln` doesn't have IIS in it, so I tried removing it. No love.

I'll ask around tomorrow and see if anyone knows what else I should try but at this point I've spent an hour and a half just trying to get this to build.

Reply:
Something seems really wrong with this solution. There are a bunch of build errors that show up every other time I build the solution.

Reply:
Building in VS can be like that sometimes. Can you build it from command line?

Reply:
If you get stuck you can still use generic host with its ConfigureHostDefaults, without ConfigureWebHostDefaults, Microsoft.AspNetCore, or IIS. You'd call ConfigureWebHost instead.